### ISSUE TITLE
Wrong shape of 'Antarctica' in 'naturalearth_lowres' dataset

### ISSUE THREAD

**Issue by iliatimofeev** [geopandas/geopandas#804]
~~~python
import geopandas as gpd

world = gpd.read_file(gpd.datasets.get_path('naturalearth_lowres'))
world[world.continent=='Antarctica'].difference(sh.geometry.box(-180,-90,180,90)).__geo_interface__
~~~
Will result:
~~~python
{'type': 'FeatureCollection',
 'features': [{'id': '6',
   'type': 'Feature',
   'properties': {},
   'geometry': {'type': 'Polygon',
    'coordinates': (((180.0, -84.71337999999997),
      (180.00000000000014, -84.71337999999999),
      (180.00000000000014, -90.00000000000003),  <--- this is not valid coordinates 
      (-179.99999999999997, -90.00000000000003), 
      (-179.99999999999997, -90.0),
      (180.0, -90.0),
      (180.0, -84.71337999999997)),)},
   'bbox': (-179.99999999999997,
    -90.00000000000003,
    180.00000000000014,
    -84.71337999999997)}],
 'bbox': (-179.99999999999997,
  -90.00000000000003,
  180.00000000000014,
  -84.71337999999997)}
~~~

original issue: https://github.com/iliatimofeev/gpdvega/issues/1

**Comment by jorisvandenbossche** [comment]
About gpdvega: that looks like a nice package! Something I need to check out

---

Do you have the same issue if you download the latest version directly from naturalearth.com: http://www.naturalearthdata.com/downloads/110m-cultural-vectors/110m-admin-0-countries/  ?

I am wondering if this is not just a floating point precision issue (what still is annoying of course, but not necessarily an "error" in the data)

A bit simpler example (without the difference and conversion to geojson):

```
In [16]: import geopandas

In [18]: df = geopandas.read_file(geopandas.datasets.get_path('naturalearth_lowres'))

In [19]: df[df.name == 'Antarctica']
Out[19]: 
   pop_est   continent        name iso_a3  gdp_md_est                                           geometry
6   3802.0  Antarctica  Antarctica    ATA       760.4  (POLYGON ((-59.57209469261153 -80.040178725096...

In [21]: antarctica = df[df.name == 'Antarctica'].geometry.squeeze()

In [24]: antarctica.bounds
Out[24]: 
(-179.99999999999997,
 -90.00000000000003,
 180.00000000000014,
 -63.27066048950458)
```

**Comment by jorisvandenbossche** [comment]
So I don't seem to have it with the latest version:

```
In [27]: df = geopandas.read_file("zip://../Downloads/ne_110m_admin_0_countries.zip")

In [33]: antarctica = df[df.NAME == 'Antarctica'].geometry.squeeze()

In [34]: antarctica
Out[34]: <shapely.geometry.multipolygon.MultiPolygon at 0x7f9a8aea2550>

In [35]: antarctica.bounds
Out[35]: (-179.99999999999994, -90.0, 180.0, -63.27066048950462)
```

Can you test if the plotting issues are also solved when using that version?

**Comment by iliatimofeev** [comment]
Yes. It helps. :)

![visualization 45](https://user-images.githubusercontent.com/11561581/44494833-10064c00-a676-11e8-8b48-678493cecb3f.png)

Would you update demo dataset?

**Comment by jorisvandenbossche** [comment]
> Would you update demo dataset?

Yes, that seems to be a good idea. I might not get directly to it (but I will tag it for the next release, so it certainly gets done), so a PR is certainly welcome as well!

**Comment by jdmcbr** [comment]
Closed by #808
